registration-processor/pre-processor/registration-processor-securezone-notification-stage/src/main/java/io/mosip/registration/processor/securezone/notification/stage/SecurezoneNotificationStage.java

```diff
@@ -279,4 +307,37 @@ public void sendMessage(MessageDTO messageDTO) {
     public MessageDTO process(MessageDTO object) {
         return null;
     }
+
+    private boolean isDuplicatePacketForSameReqId(MessageDTO messageDTO) {
+        // Validate if duplicate packet received against same additional_info_req_id
+        SyncRegistrationEntity entity = syncRegistrationService.findByWorkflowInstanceId(messageDTO.getWorkflowInstanceId());
+        // find all records for same additionalInfoReqId.
+        List<SyncRegistrationEntity> entities = syncRegistrationService.findByAdditionalInfoReqId(entity.getAdditionalInfoReqId());
+        // if multiple records are present for same additionalInfoReqId then check in registration table how many packets are received
+        if (!CollectionUtils.isEmpty(entities) && entities.size() > 1) {
+            List<String> workflowInstanceIds = entities.stream().map(e -> e.getWorkflowInstanceId()).collect(Collectors.toList());
+            List<InternalRegistrationStatusDto> dtos = new ArrayList<>();
+            for (String w : workflowInstanceIds) {
+                InternalRegistrationStatusDto dto = registrationStatusService.getRegistrationStatus(
+                        messageDTO.getRid(), messageDTO.getReg_type(), messageDTO.getIteration(), w);
+                if (dto != null )
```

Comment:
message DTO anyways contains rid, reg type and iteration, can we just directly query registration table for all matches sorted by create date time asc, if the current workflow Id is the first one then we will say isValid as true else false ?

Reply:
first i am getting all the workflowInstanceId for same additional info req id.
then i get the records in registration table for those workflowInstanceId
then i check datetime on records of registration table
so if packet is uploaded only then record will be present in registration table